processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -74,7 +88,16 @@ func (d *detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 	d.rb.SetCloudProvider(conventions.AttributeCloudProviderAWS)
 	d.rb.SetCloudPlatform(conventions.AttributeCloudPlatformAWSEKS)
 
-	return d.rb.Emit(), conventions.SchemaURL, nil
+	// The error is unhandled because we want to return successfully detected resources
+	// regardless of an error. The caller will properly handle any error hit while getting
+	// the cluster name.
+	if d.ra.K8sClusterName.Enabled {
+		var clusterName string
+		clusterName, err = d.utils.getClusterName(ctx)
+		d.rb.SetK8sClusterName(clusterName)
+	}
+
+	return d.rb.Emit(), conventions.SchemaURL, err
```

Comment:
Who is the caller in this case and how is it handled? Is the caller built in such a way where it treats this as a partial error and doesn't immediately exit on error?

Reply:
Good point, caller is here:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b716a4d89acd497cb1ed044b941c49a7897b72bb/processor/resourcedetectionprocessor/internal/resourcedetection.go#L129-L134

The error will result in the correctly detected resources not being returned. I'll add a warning log message instead of returning an error here, I believe that's the best way to balance this.

Reply:
Tested with update, here's the logging output:
```
2023-11-09T23:03:04.703Z	info	internal/resourcedetection.go:125	began detecting resource information	{"kind": "processor", "name": "resourcedetection/eks", "pipeline": "logs"}
2023-11-09T23:03:07.846Z	warn	eks/detector.go:97	Unable to get EKS cluster name	{"kind": "processor", "name": "resourcedetection/eks", "pipeline": "logs", "error": "EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestCanceled: request context canceled\ncaused by: context deadline exceeded"}
2023-11-09T23:03:07.846Z	info	internal/resourcedetection.go:139	detected resource information	{"kind": "processor", "name": "resourcedetection/eks", "pipeline": "logs", "resource": {"cloud.platform":"aws_eks","cloud.provider":"aws"}}
```
We are able to receive successfully detected resources, with the log message for the `k8s.cluster.name` failure.